core/trashcan.js

```diff
@@ -510,14 +521,38 @@ Blockly.Trashcan.prototype.click = function() {
     xml[i] = Blockly.Xml.textToDom(text);
   }
   this.flyout.show(xml);
+
+  this.fireUiEvent_();
+};
+
+/**
+ * Fires a ui event for trashcan click.
+ * @private
+ */
+Blockly.Trashcan.prototype.fireUiEvent_ = function() {
+  var uiEvent = new Blockly.Events.Ui(null, 'trashcan', null, null);
```

Comment:
I originally had "trashcanOpen" to follow the pattern of "commentOpen" and "mutatorOpen". However, using the naming "trashcan" is consistent with "category" ui event, which also opens flyout and I'm also not sure we want to emit an event every time a flyout is closed.

Reply:
I've updated back to 'trashcanOpen', but I'm open to discussion.